add_me_to_your_PATH/Dockerfile.template

```diff
@@ -102,41 +145,24 @@ RUN apt-get -qq update && \
 ENV JULIA_PROJECT /JuliaProject
 COPY --from=sysimage-project Project.toml Manifest.toml ${JULIA_PROJECT}/
 
-RUN --mount=type=secret,id=github_token \
+RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache \
+    --mount=type=secret,id=github_token \
     julia -e 'using Pkg; Pkg.Registry.update(); Pkg.instantiate(); Pkg.build(); Pkg.precompile(strict=true)'
 
+# Install required dependencies for sysimage.jl
+RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache \
+    --mount=type=secret,id=github_token \
+    julia -e 'using Pkg; Pkg.add(PackageSpec(name="PackageCompiler", version="1"); preserve=Pkg.PRESERVE_ALL)'
```

Comment:
is this gonna install this dependency in teh project environment or the global environment?  seems like the global environment would be safer but setting `JULIA_PROJECT` means that by default it'll use the JuliaProject dir from the base image right?